### DISCUSSION TITLE
How do I run the "NPDE" function with Pmetrics, version 1.9.7?

### DISCUSSION THREAD

**Discussion by Mannny-PK** [LAPKB/Pmetrics#111] [Q&A]
Dear Professor,
I have encountered a problem when I used Pmetrics, version 1.9.7，How do I run the "NPDE" function?
When I ran the following code, it failed.

setwd(paste(wd,"/Runs",sep=""))
npde.2 <- makeNPDE(run=2,limits=NA)
str(npde.2)
npde.2[[1]]@results@res
plot(npde.2)
plot(npde.2,plot.type="vpc",ylog=F)
npc.2 <- plot(sim.2,obs=op.2,log=F)
npc.2$npc

**Comment by mnneely** [comment]
That code has been deprecated. The current code is makeValid. Also, we have released v. 2.0 of Pmetrics. You should consider updating via devtools::install_github(“LAPKB/Pmetrics”). Documentation for all the changes in 2.0 (and there are many) is here: https://lapkb.github.io/Pmetrics/.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Mannny-PK ***@***.***>
Sent: Friday, November 11, 2022 12:48:13 AM
To: LAPKB/Pmetrics ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [LAPKB/Pmetrics] How do I run the "NPDE" function with Pmetrics, version 1.9.7? (Discussion #111)


Dear Professor,
I have encountered a problem when I used Pmetrics, version 1.9.7，How do I run the "NPDE" function?
When I ran the following code, it failed.

setwd(paste(wd,"/Runs",sep=""))
npde.2 <- makeNPDE(run=2,limits=NA)
str(npde.2)
***@***.******@***.***
plot(npde.2)
plot(npde.2,plot.type="vpc",ylog=F)
npc.2 <- plot(sim.2,obs=op.2,log=F)
npc.2$npc

**Reply by Mannny-PK** [reply]
Great thanks for your advice! Where can I find the script for the current code "makeValid"?

**Reply by Johnny0514** [reply]
Have you resolved these problems? Could you please tell me how to use "make_valid"?
